ui/tests/e2e/search_log.test.ts

```diff
@@ -22,32 +22,62 @@ describe('Search Logs', () => {
       await ppExpect(page).toClick('button#signin_btn')
 
       // jump to search logs page
-      await ppExpect(page).toClick('a#search_logs')
+      await page.waitForSelector('a#search_logs')
+      const searchLogsLink = await page.$('a#search_logs')
+      await searchLogsLink.click()
+
+      // this fails randomly and high possibility, says can't find "a#search_logs" element
+      // await ppExpect(page).toClick('a#search_logs')
 
       // find search form
       const searchForm = await page.waitForSelector('form#search_form')
 
+      // choose time range
+      await ppExpect(searchForm).toClick(
+        'button[data-e2e="timerange-selector"]'
+      )
+      await ppExpect(page).toClick(
+        'div[data-e2e="common-timeranges"] div:last-child'
+      )
+      await ppExpect(searchForm).toClick(
+        'button[data-e2e="timerange-selector"]'
+      )
+
       // set log level to INFO
-      await ppExpect(searchForm).toClick('#log_level_selector')
-      await ppExpect(page).toClick('div[data-e2e="level_info"]')
+      await ppExpect(searchForm).toClick('#logLevel')
+      await ppExpect(page).toClick('div[data-e2e="level_2"]')
 
       // select TiDB component
-      await ppExpect(searchForm).toClick('div[data-e2e="components_selector"]')
+      // https://stackoverflow.com/questions/59882543/how-to-wait-for-a-button-to-be-enabled-and-click-with-puppeteer
+      await page.waitForSelector('div#instances input:not([disabled])')
+      await ppExpect(searchForm).toClick('div#instances')
       // components selector dropdown is a DOM node with absolute position
       // and its parent is body, failed to add id or data-e2e to it
-      await ppExpect(page).toClick('span[title="TiDB"]')
+      // cancel select PD and TiKV, and only remain TiDB
+      await ppExpect(page).toClick('span', { text: 'PD' })
```

Comment:
a span with text 'PD' may occur very frequently. Can we make it more accurate?

Reply:
I think so, let me try it.

Reply:
Not easy to add additional props to the `span` directly, so change to add `data-e2e="table-with-filter"` to its container. The matching changes to:

```
await ppExpect(page).toClick('div[data-e2e="table-with-filter"] span', {
    text: 'PD',
})
```